source/daplink/interface/target_flash.c

```diff
@@ -253,15 +307,31 @@ static error_t target_flash_erase_chip(void)
 {
     if (g_board_info.target_cfg){
         error_t status = ERROR_SUCCESS;
-        const program_target_t *const flash = g_board_info.target_cfg->flash_algo;
-
-        status = flash_func_start(FLASH_FUNC_ERASE);
-        if (status != ERROR_SUCCESS) {
-            return status;
-        }
-        if (0 == swd_flash_syscall_exec(&flash->sys_call_s, flash->erase_chip, 0, 0, 0, 0)) {
-            return ERROR_ERASE_ALL;
-        }
+        program_target_t * flash;
+        region_info_t * flash_region = g_board_info.target_cfg->extra_flash;
+        
+        do { 
+            flash = current_flash_algo;
```

Comment:
I think you need to set `flash` to the main flash algo here, not the current one. Otherwise it's possible to miss some regions.

And you know, I'm really wondering if we should just merge all the flash and ram regions into a single array of each, instead of having a main region and then extra regions array. It would make this code simpler. But don't worry about that for now, maybe later…